### PR TITLE
📁 Fill Perf Tuning default config with accelerator spec

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
This pull request includes changes to improve testing and mocking in `test/unit_test/passes/onnx/test_perf_tuning.py` and add support for specifying device and execution provider in the default configuration of the ONNX pass in `olive/passes/onnx/perf_tuning.py`.

Testing improvements:

* <a href="diffhunk://#diff-3c4b1c200e2ac992cd17d17e72a48402d7376e0dcc11d3c4a3728d630ec48097L14-R35">`test/unit_test/passes/onnx/test_perf_tuning.py`</a>: Added additional test cases and import statements to improve testing and mocking. <a href="diffhunk://#diff-3c4b1c200e2ac992cd17d17e72a48402d7376e0dcc11d3c4a3728d630ec48097L14-R35">[1]</a> <a href="diffhunk://#diff-3c4b1c200e2ac992cd17d17e72a48402d7376e0dcc11d3c4a3728d630ec48097R5">[2]</a>

Pass improvements:

* <a href="diffhunk://#diff-01a5090bdcf3c820e5075bea71814779996751d98719a0b2f0c92c77445b7760R281-R283">`olive/passes/onnx/perf_tuning.py`</a>: Added support for specifying device and execution provider in the default configuration of the ONNX pass, and removed redundant code for setting default values. <a href="diffhunk://#diff-01a5090bdcf3c820e5075bea71814779996751d98719a0b2f0c92c77445b7760R281-R283">[1]</a> <a href="diffhunk://#diff-01a5090bdcf3c820e5075bea71814779996751d98719a0b2f0c92c77445b7760L306-R310">[2]</a> <a href="diffhunk://#diff-01a5090bdcf3c820e5075bea71814779996751d98719a0b2f0c92c77445b7760R5">[3]</a> <a href="diffhunk://#diff-01a5090bdcf3c820e5075bea71814779996751d98719a0b2f0c92c77445b7760L323-R327">[4]</a> <a href="diffhunk://#diff-01a5090bdcf3c820e5075bea71814779996751d98719a0b2f0c92c77445b7760L351-L358">[5]</a>

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

**Review by jambayk — DISMISSED** [review]

File: test/unit_test/passes/onnx/test_perf_tuning.py [hunk 1]
```diff
@@ -22,6 +26,36 @@ def test_ort_perf_tuning_pass(config, tmp_path):
...
+    # setup
+    input_model = get_onnx_model()
+    p = create_pass_from_dict(OrtPerfTuning, config, disable_search=True)
+
+    # execute
+    p.run(input_model, None, None)
+
+    # assert
+    if "providers_list" not in config:
+        assert mock_run.call_args.args[2]["providers_list"] == ["CPUExecutionProvider"], (
```
Comment (guotuofeng) [inline test/unit_test/passes/onnx/test_perf_tuning.py:47]: shouldn't the ep list == ["CPUExecutionProvider", "CUDAExecutionProvider"]?

File: test/unit_test/passes/onnx/test_perf_tuning.py [hunk 2]
```diff
@@ -22,6 +26,36 @@ def test_ort_perf_tuning_pass(config, tmp_path):
...
+            "providers_list is not set correctly as ['CPUExecutionProvider'] by default when"
+            " user does not specify it"
+        )
+    if "device" not in config:
+        assert (
+            mock_run.call_args.args[2]["device"] == "cpu"
+        ), "device is not set correctly as cpu by default when user does not specify it"
```
Comment (guotuofeng) [inline test/unit_test/passes/onnx/test_perf_tuning.py:54]: should it be clear?
if config:
     xxx
else:
    for k, v in config.items():

**Review by guotuofeng — APPROVED** [review]

File: test/unit_test/passes/onnx/test_perf_tuning.py (hunk 2 above)
Comment (trajepl) [reply test/unit_test/passes/onnx/test_perf_tuning.py:54]: There is the case config is not None but `providers_list` and `device` are not set.
The `providers_list` and `device` are required no matter the config is none or not.

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**